src/main/java/thrift/MainApp.java

```diff
@@ -160,8 +161,22 @@ protected void initCurrencyMappings(Storage storage) {
             logger.warning("Currency Mappings file at " + currencyMappingsFilePath
                     + " is not in the correct format. "
                     + "Using default currency mappings");
+
+            Map<String, Double> defaultCurrencyMappings = CurrencyUtil.getCurrencyMap();
+            try {
+                storage.saveCurrencyMappings(defaultCurrencyMappings);
+            } catch (IOException ioe) {
+                //do nothing
+            }
         } catch (IOException | NoSuchElementException e) {
             logger.warning("Problem while reading from the file. Using default currency mappings");
+
+            Map<String, Double> defaultCurrencyMappings = CurrencyUtil.getCurrencyMap();
+            try {
+                storage.saveCurrencyMappings(defaultCurrencyMappings);
+            } catch (IOException ioe) {
+                //do nothing
+            }
```

Comment:
Repeated code, should do some form of abstraction 😅

Reply:
The logic has been changed to save the file regardless of whether a valid file exists or not. The code no longer requires repetition.